Make distro conditional more flexible

gotmax23 opened the pull request:
- Regenerate README
- Make distro conditional more flexible

**Describe the change**
I changed `ansible_distribution in [ "CentOS", "RedHat", "Rocky" ]` >
`ansible_os_family == "RedHat"`. I did this after I noticed this role
doesn't work on Almalinux. I think this is better than just adding
`Almalinux` to the array, as it makes this role automatically compatible
with all RHEL clones without us having to list them manually.

gotmax23 committed:
Regenerate README

gotmax23 committed:
Make distro conditional more flexible

I changed `ansible_distribution in [ "CentOS", "RedHat", "Rocky" ]` >
`ansible_os_family == "RedHat"`. I did this after I noticed this role
doesn't work on Almalinux. I think this is better than just adding
`Almalinux` to the array, as it makes this role automatically compatible
with all RHEL clones without us having to list them manually.

robertdebock commented:
(Correct me if I'm wrong please.)

`ansible_os_family` would include Fedora, which can't have EPEL installed, right?

gotmax23 committed:
Fix `role-name` linting error

gotmax23 committed:
Fix Rocky Linux

gotmax23 commented:
> (Correct me if I'm wrong please.)
> 
> `ansible_os_family` would include Fedora, which can't have EPEL installed, right?

Yes, `ansible_os_family == "RedHat"` includes Fedora. However, the `ansible_distribution_major_version in [ "7", "8" ]` condition should exclude Fedora (Both Fedora 7 and 8 were released in 2007!). The idea was to make this role as compatible as possible.

It seems the tests have failed on Rocky Linux because of the incorrect `ansible_os_family` value issue that I explained [here](https://github.com/robertdebock/ansible-role-bootstrap/commit/da3eb39e640a4ec63a95cfe633f0542dc1b7ea7b#commitcomment-53196938) in our conversation about the CentOS 8 early EOL. I fixed this in my most recent commit. I also fixed a linting error for you.

If you really don't like this change, we can change `ansible_distribution in [ "CentOS", "RedHat", "Rocky" ]` to `ansible_distribution in [ "CentOS", "RedHat", "Rocky", "AlmaLinux", "OracleLinux"]` (Those are the missing distributions off the top of my head).

Thanks,
Maxwell 😀

robertdebock commented:
That looks good. Why did I ever use `ansible_role` here? Well, thanks for spotting that.

I'll wait for CI to pass and merge.
